### ISSUE TITLE
nixpkgs unstable: non-existing theharvester version

### ISSUE THREAD

**Issue by blshkv** [repology/repology#560]
https://repology.org/metapackage/theharvester/versions
v2.7.1 hasn't been released yet

**Comment by blshkv** [comment]
somehow, this is not fixed completely.
Kali came up with a 2.7.2 snapshot  (http://git.kali.org/gitweb/?p=packages/theharvester.git) and the package is listed as outdated again:
https://repology.org/metapackage/theharvester/versions

**Comment by AMDmi3** [comment]
No fake versions are ever 'fixed completely', as there's no way telling fake version appearing in another repository from an actual release. Nix seem to have huge update with thousands of new packages and seemingly a handful of incorrect versions, I'll only be able to properly investigate this in a week.

**Comment by globin** [comment]
The new packages are our set of haskell/R packages which weren't in the generated JSON earlier (https://github.com/NixOS/nixos-homepage/commit/4c5bcc135cb0ffcc68941c699bcf85c74138836f)

**Comment by AMDmi3** [comment]
@globin it's good you're here. Is there a way to differentiate them? They seem to generate a lot of conflicts with other packages.

**Comment by globin** [comment]
The attribute name (our unique identifier, key in the json file) is `haskellPackages.${name}` or respectively `rpackages.${name}` and for those the name in the json is prefixed with `r-`:
```
...
  "haskellPackages.activitystreams-aeson": {                                                                                   
    "name": "activitystreams-aeson-0.2.0.2",                      
...
  "rPackages.zyp": {                                                                                                           
    "name": "r-zyp-0.10-1",                                                                                                                                                                
```

**Comment by AMDmi3** [comment]
Thanks, that should be enough. As said, will handle this in a week.

**Comment by blshkv** [comment]
Could you also fix kali rolling and parrot versioning. Debian-version-date is a snapshot, not a release.

**Comment by AMDmi3** [comment]
Regarding theHarvester, the original issue turns out to not be completely valid, as `2.7.[12]` versions are mentioned in the changelog, there just are no tags. I think the least painful way to fix this is to masquerade `2.7.[12]` as `2.7`. This will break again when `2.7.3` is released, but I hope that theHarverster author will take laramies/theHarvester#82 into account and tag it properly.

**Comment by blshkv** [comment]
that's actually a bit of speculation but ok. @laramies seems doesn't care about a proper maintenance.
Thanks!
